wireapp/wire-android
# fix: various issues with Android 13 and permissions [AR-2735]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2735" title="AR-2735" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />AR-2735</a>  Permission dialogs on Android 13
  </td></table>
  <br />
 






----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

# What's new in this PR?

### Issues
After bumping our app's target API to Android 13, it caused some side effects. Namely:
1. Starting from Tiramisu version, `READ_EXTERNAL_STORAGE` permission got deprecated and it is not needed to be requested unless user wants to read other image, video or audio that has actually been created and only owned by other apps. For now, we are fine since all these images are already owned by the default us.
2. After targetting Tiramisue version, now the permission to show android notifications must be requested in the same way that other permissions. 


### Solutions

Adapted to Android 13 new permission requirements.


#### Test Coverage (Optional)


#### How to Test

Start the app and make sure that images and files can be shared, and that android push notifications are actually arriving to the device.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.
